csrc/cache_kernels.cu

```diff
@@ -185,13 +186,34 @@ __global__ void reshape_and_cache_kernel(
                                   + head_idx * head_size * block_size
                                   + head_offset * block_size
                                   + block_offset;
-    key_cache[tgt_key_idx] = key[src_key_idx];
-    value_cache[tgt_value_idx] = value[src_value_idx];
+    scalar_t tgt_key = key[src_key_idx];
+    scalar_t tgt_value = value[src_value_idx];
+    if constexpr (enable_fp8_kv_cache) {
```

Comment:
We will have more different types of quantization in the future, so let's try to make this a string or enum instead of a bool. And set its value to `fp8_e5m2`.